src/CMakeLists.txt

```diff
@@ -246,17 +230,151 @@ ADD_DEFINITIONS(-DEIGEN_MPL2_ONLY)
 # - external if CERES_DIR_HINTS and FIND_PACKAGE return a valid Ceres setup
 # ==============================================================================
 FIND_PACKAGE(Ceres QUIET HINTS ${CERES_DIR_HINTS})
-IF(Ceres_FOUND)
-   MESSAGE("CERES FOUND")
-ELSE()
-  MESSAGE("CERES NOT FOUND")
+IF(NOT Ceres_FOUND)
   SET(OpenMVG_USE_INTERNAL_CERES ON)
   SET(CERES_INCLUDE_DIRS
     ${CMAKE_CURRENT_SOURCE_DIR}/third_party/ceres-solver/include
     ${CMAKE_CURRENT_SOURCE_DIR}/third_party/ceres-solver/internal/ceres/miniglog
     ${PROJECT_BINARY_DIR}/third_party/ceres-solver/config)
+    FILE(READ "${CMAKE_CURRENT_SOURCE_DIR}/third_party/ceres-solver/VERSION" CERES_CONFIG)
+    STRING(REGEX REPLACE "version ([0-9.]+).*" "\\1" CERES_VERSION ${CERES_CONFIG})
   SET(CERES_LIBRARIES ceres cxsparse)
-ENDIF(Ceres_FOUND)
+ENDIF()
+
+# ==============================================================================
+# Flann
+# ==============================================================================
+# - internal by default (flann),
+# - external if FLANN_DIR_HINTS and a valid Flann setup is found
+# ==============================================================================
+FIND_PATH(FLANN_INCLUDE_DIRS flann.hpp ${FLANN_DIR_HINTS})
```

Comment:
@pmoulon ,

  An now i realize that going pkg-config way will miss the nice VERSION feature. So lets better agree with intial No.2 proposal and do one FindX.cmake for each one library since VERSION can be extracted quite different ways for each lib so universal FindX.cmake will be not feasable at some point.
 
 I now start to see clear advantages for having one FindX.cmake for each lib, and deal them in slight different way.